extensions/slack/src/shared/client/SlackClient.ts

```diff
@@ -0,0 +1,162 @@
+import { slackWebClient } from "./WebClient";
+import { LocalStorageService } from "./LocalStorageService";
+
+interface Item {
+  id: string;
+  teamId: string;
+  name: string;
+}
+
+export interface User extends Item {
+  username: string;
+  icon: string | undefined;
+}
+
+export type Channel = Item;
+export type Group = Item;
+
+export type PresenceStatus = "online" | "offline" | "forced-offline";
+export interface SnoozeStatus {
+  nextDndEnd: Date | undefined;
+  snoozeEnd: Date | undefined;
+}
+
+const sortNames = (a: string, b: string) => {
+  if (a < b) {
+    return -1;
+  }
+  if (a > b) {
+    return 1;
+  }
+  return 0;
+};
+
+export class SlackClient {
+  public static async getUsers(): Promise<User[]> {
+    const users = await LocalStorageService.get<User[]>("users", async (): Promise<User[]> => {
+      const userList = await slackWebClient.users.list();
+
+      const result =
+        userList.members
+          ?.filter(
+            ({ is_bot, is_workflow_bot, deleted, id }) => !is_bot && !is_workflow_bot && !deleted && id !== "USLACKBOT"
+          )
+          .map(({ id, name: username, profile, team_id }) => {
+            const firstName = profile?.first_name ?? "";
+            const lastName = profile?.last_name ?? "";
+            const name = `${firstName} ${lastName}`;
+
+            const displayName = name.trim() ? name : profile?.display_name;
+
+            return {
+              id,
+              name: displayName,
+              icon: profile?.image_24,
+              teamId: team_id,
+              username,
+            };
+          })
+          .filter((i): i is User => (i.id && i.id.trim() && i.name?.trim() && i.teamId?.trim() ? true : false))
+          .sort((a, b) => sortNames(a.name, b.name)) ?? [];
+
+      return result;
+    });
+
+    return users ?? [];
+  }
+
+  public static async getChannels(): Promise<Channel[]> {
+    const channels = await LocalStorageService.get<Channel[]>("channels", async (): Promise<Channel[]> => {
+      const publicChannels = await slackWebClient.conversations.list({
+        exclude_archived: true,
+        types: "public_channel",
+        limit: 1000,
+      });
+      const privateChannels = await slackWebClient.conversations.list({
+        exclude_archived: true,
+        types: "private_channel",
+        limit: 1000,
+      });
```

Comment:
`conversations.list` accepts `types` as comma separated list of types ([see conversations.list#arg_types](https://api.slack.com/methods/conversations.list#arg_types)), so there is no need to make two requests:
```suggestion
      const channels = await slackWebClient.conversations.list({
        exclude_archived: true,
        types: "public_channel, private_channel",
        limit: 1000,
      });
```
What's more, you can provide all of these:
```
types: "public_channel, private_channel, mpim"
```
So you could have one single request to get private/public channels and groups. Then you can filter whichever needs to be displayed.